### PR TITLE
D2IQ-69336: E2E Tests

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
**What type of PR is this?**

Testing

**What this PR does/ why we need it**:

Adds UI based E2E tests to PRs. 
These tests spin up a cluster with the PR version of this repo and use the UI to test the kommander functionality

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-69794

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: Dispatchfile [hunk 1]
```diff
@@ -50,5 +61,76 @@ dindTask("dispatch-integration-test",
...
+                                       valueFrom=secretVar("rotator-ksphere-kommander",
+                                                           "AWS_ACCESS_KEY_ID")),
+                     k8s.corev1.EnvVar(name="AWS_SECRET_ACCESS_KEY",
+                                       valueFrom=secretVar("rotator-ksphere-kommander",
+                                                           "AWS_SECRET_ACCESS_KEY")),
+                     k8s.corev1.EnvVar(name="LICENSE",
+                                       valueFrom=secretVar("kommander-license",
+                                                           "LICENSE"))
+                 ],
+                 command=["bash", "./test/scripts/e2e.sh", "$(context.git.branch)"]),
```
Comment (shaneutt) [inline Dispatchfile:130]: It looks like this _always_ runs, I just want to verify that was the intentional _and_ request some additional comments in the PR description to further explain and provide information about what this means to E2E test run times going forward, e.t.c. :thinking:

Comment (DanielMSchmidt) [reply Dispatchfile:130]: Yeah, the idea was to have them always run so that we can catch problems early on.

Comment (DanielMSchmidt) [reply Dispatchfile:130]: What kind of information are you looking for?

Comment (shaneutt) [reply Dispatchfile:130]: I'll unblock, just curious how long the runs take?

Comment (DanielMSchmidt) [reply Dispatchfile:130]: They take around 30-45 minutes. Creating the konvoy cluster itself takes as long as running the tests

**Comment by DanielMSchmidt** [comment]
@shaneutt Seems like they take over an hour on CI

File: Dispatchfile (hunk 1 above)
Comment (joejulian) [reply Dispatchfile:130]: I think there may be a disconnect here. I think Shane means "always" as in even if addons that don't affect the UI are changed where I think Daniel meant "always" in opposition to "nightly".

Comment (DanielMSchmidt) [reply Dispatchfile:118]: Ah I see. We talked about filtering the tests at some point, but tbh that seems like an over optimization right now. Creating the initial cluster and creating the clusters through kommander take time, testing the addons is currently relatively quick (all in < 5 min)

**Comment by shaneutt** [comment]
Note: you'll want to rebase to latest master, which includes general CI fixes.

**Comment by shaneutt** [comment]
@DanielMSchmidt looks like there are some CI issues: if the @mesosphere/sig-ksphere-catalog team can provide any assistance for you getting this completed, please let us know!

**Comment by DanielMSchmidt** [comment]
@shaneutt Yeah, I think the main problem is that the E2E tests are not run on the UI repo and therefore the UI and the tests go out of sync. I'm currently working on getting them working in the UI repo and after that I will continue the work on this PR.

File: Dispatchfile [hunk 2]
```diff
@@ -10,10 +10,55 @@ load("github.com/mesosphere/dispatch-tasks/bump_charts/bump_charts@master", "bum
...
+resource("artifacts", type="storage", params={
+    "type": "gcs",
+    "location": "s3://artifacts",
+    "dir": "yes"
+}, secrets={
+    "BOTO_CONFIG": k8s.corev1.SecretKeySelector(key="boto", localObjectReference=k8s.corev1.LocalObjectReference(name="s3-config"))
+})
+
 git_resource(git_clone_dirname, url="$(context.git.url)", revision="$(context.git.commit)")
+git_resource("ui-git", url="git@github.com:mesosphere/kommander.git", revision="v6.35.0")
```
Comment (juliangieseke) [inline Dispatchfile:22]: how does this get updated? manually by UI team?

File: Dispatchfile [hunk 3]
```diff
@@ -10,10 +10,55 @@ load("github.com/mesosphere/dispatch-tasks/bump_charts/bump_charts@master", "bum
...
+            workingDir="/workspace/" + git_clone_dirname,
+            args=["git", "fetch", "origin", "master"]),
+
+        v1.Container(
+            name="run-e2e-tests",
+            image="mesosphere/kommander-ci:1045-0bc7580",
+            workingDir="/workspace/" + git_clone_dirname,
+            resources=k8s.corev1.ResourceRequirements(
+                limits={
+                    "cpu": k8s.resource_quantity("8000m"),
```
Comment (juliangieseke) [inline Dispatchfile:42]: 😳

File: test/scripts/e2e.sh [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+KOMMANDER_INSTALL_PID=$!
+cd "$KOMMANDER_REPO_PATH/system-tests"
+npm install >"$OUTPUT_PATH/system-test-install.log" 2>&1 &
+INSTALL_PID=$!
+
+cd "$PROJECT_ROOT"
+
+# Set up Konvoy
+echo "Setup Konvoy"
+source ${PROJECT_ROOT}/test/scripts/setup-konvoy.sh v1.5.0
```
Comment (juliangieseke) [inline test/scripts/e2e.sh:79]: how do we keep this up to date?

File: test/scripts/e2e.sh [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        - name: reloader
+          enabled: true
+        - name: traefik
+          enabled: true
+        - name: traefik-forward-auth
+          enabled: true
+        - name: velero
+          enabled: true
+    - configRepository: https://github.com/mesosphere/kubeaddons-conductor
+      configVersion: stable-1.17-1.0.0
```
Comment (juliangieseke) [inline test/scripts/e2e.sh:254]: this probably should be master 🤔 
```suggestion
      configVersion: master
```

File: test/scripts/e2e.sh [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+      addonsList:
+        - name: conductor
+          enabled: false
+    - configRepository: https://github.com/mesosphere/kubeaddons-dispatch
+      configVersion: stable-1.17-1.2.2
```
Comment (juliangieseke) [inline test/scripts/e2e.sh:259]: this probably should be master 🤔 
```suggestion
      configVersion: master
```

File: test/scripts/e2e.sh [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+      addonsList:
+        - name: dispatch
+          enabled: false
+    - configRepository: https://github.com/mesosphere/kubeaddons-kommander
+      configVersion: stable-1.17-1.1.0
```
Comment (juliangieseke) [inline test/scripts/e2e.sh:264]: this for sure should be master 
```suggestion
      configVersion: master
```

File: test/scripts/e2e.sh [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+      addonsList:
+        - name: kommander
+          enabled: true
+  version: v1.5.0
```
Comment (juliangieseke) [inline test/scripts/e2e.sh:268]: how do we keep this up to date?

**Review by juliangieseke — COMMENTED** [review]
Im pretty sure devprod can help with all these "keep things up to date" questions.

File: Dispatchfile (hunk 2 above)
Comment (DanielMSchmidt) [reply Dispatchfile:22]: yes, when a new UI version is bumped or when a new test is merged we need to bump this

File: Dispatchfile (hunk 3 above)
Comment (DanielMSchmidt) [reply Dispatchfile:42]: Maybe that's a little too much, but I'd rather have it green first before optimizing.

File: test/scripts/e2e.sh (hunk 4 above)
Comment (DanielMSchmidt) [reply test/scripts/e2e.sh:79]: By making a PR :)

File: test/scripts/e2e.sh (hunk 5 above)
Comment (DanielMSchmidt) [reply test/scripts/e2e.sh:254]: From my perspective no. We want to just test the changes in this repo but not master changes on the others, right?

File: test/scripts/e2e.sh (hunk 8 above)
Comment (DanielMSchmidt) [reply test/scripts/e2e.sh:268]: Manually unfortunately. It's a tradeoff between this and using konvoy init and customizing this through tooling

File: test/scripts/e2e.sh (hunk 5 above)
Comment (juliangieseke) [reply test/scripts/e2e.sh:254]: no, we want to test the changes on this _PR_ against the next (to be released) version of the whole system. therefore `master` branches is what we want to test. (and PR branch for the repo under test)

**Review by juliangieseke — APPROVED** [review]

**Comment by juliangieseke** [comment]
> The base branch requires all commits to be signed

ping @DanielMSchmidt

**Comment by DanielMSchmidt** [comment]
@shaneutt This one is ready now, but the integration tests are red, any idea why that could be?

**Comment by shaneutt** [comment]
> @shaneutt This one is ready now, but the integration tests are red, any idea why that could be?

Wait for TeamCity, we still have a fair number of problems with dispatch and it's not required. If TeamCity fails as well, I'll dig in.

**Comment by DanielMSchmidt** [comment]
@shaneutt Teamcity says no

Computer says no

**Comment by joejulian** [comment]
ginkgo timed out

Is this test even run in teamcity?

**Comment by DanielMSchmidt** [comment]
@joejulian no the new tests are run in dispatch and nothing here should change anything with teamcity. Pretty sure it must be flake

**Comment by joejulian** [comment]
@shaneutt I don't think that's true, anymore, that there are any problems with the dispatch tests. They've been passing consistently and we have it on the agenda to retire TC in favor of Dispatch so I, personally, don't feel we should just dismiss that out-of-hand.

**Comment by joejulian** [comment]
test

**Comment by shaneutt** [comment]
It looks like with this change, we will need to hereafter make dispatch a _required_ test.

**Comment by DanielMSchmidt** [comment]
/test-e2e

Seems like an infra / product flake, nodes could not be reached after konvoy up

**Pull request merged**